src/main/java/ch/wisv/areafiftylan/service/TicketServiceImpl.java

```diff
@@ -84,18 +85,26 @@ public TicketTransferToken setupForTransfer(Long ticketId, String goalUserName){
         User u = userService.getUserByUsername(goalUserName).orElseThrow(() -> new UsernameNotFoundException("User " + goalUserName + " not found."));
         Ticket t = ticketRepository.findOne(ticketId);
 
-        TicketTransferToken ttt = new TicketTransferToken(u, t);
+        List<TicketTransferToken> ticketTransferTokens = tttRepository.findAllByTicketId(ticketId).stream()
+                .filter(Token::isValid)
+                .collect(Collectors.toList());
 
-        tttRepository.save(ttt);
+        if (!ticketTransferTokens.isEmpty()) {
+            throw new IllegalStateException("Ticket " + ticketId + " is already set up for transfer!");
```

Comment:
I would think about making a special exception for this. This is more in line with our programming style and makes sure the exceptions (with the message strings) are nicely bundled in the exceptions folder instead of scrambled in all the code.